modular_skyrat/master_files/code/modules/client/preferences/headshot.dm

```diff
@@ -0,0 +1,37 @@
+/datum/preference/text/headshot
+	category = PREFERENCE_CATEGORY_NON_CONTEXTUAL
+	savefile_identifier = PREFERENCE_CHARACTER
+	savefile_key = "headshot"
+	/// Stored headshot link from `value` to cut down on chat spam
+	var/stored_link
+
+/datum/preference/text/headshot/apply_to_human(mob/living/carbon/human/target, value, datum/preferences/preferences)
+	target.dna.features["headshot"] = value
+
+/datum/preference/text/headshot/is_valid(value)
+	if(!length(value)) //Just to get blank ones out of the way
+		return TRUE
+	if(!is_veteran_player(usr?.client) && !(usr?.ckey in GLOB.donator_list))
+		to_chat(usr, span_warning("You must be a veteran or donator to use this feature!"))
+		return
+	if(!findtext(value, "https://"))
+		to_chat(usr, span_warning("You need \"https://\" in the link!"))
+		return
+	if(!findtext(value, ".png") && !findtext(value, ".jpg"))
+		to_chat(usr, span_warning("You need either \".png\" or \".jpg\" in the link!"))
+		return
+	if(!findtext(value, "imgur") && !findtext(value, "gyazo") && !findtext(value, "media.discordapp"))
+		to_chat(usr, span_warning("The link needs to be an unshortened Imgur, Gyazo, or Discordapp link!"))
+		return
```

Comment:
i can literally host a file of my own, in my own domain, as bigchungusgyazo.png.html and it will be loaded.

Reply:
Resolved